services/sources/publisher_backlist_service.py

```diff
@@ -23,6 +24,40 @@ def __init__(self):
         
         self.airtable_auth_token = os.environ.get('AIRTABLE_KEY', None)
 
+    def delete_records(
+        self,
+        limit: Optional[int]=None
+    ):
+        filter_by_formula = self.build_filter_by_formula_parameter(deleted=True)
+        
+        array_json_records = self.get_records_array(limit, filter_by_formula)
+
+        for json_dict in array_json_records:
+            if json_dict['records']:
+                for records_value in json_dict['records']:
+                    try:
```

Comment:
I think because we have the try/except login below in delete_manifeset - we can remove the try/except block here. You can add some extra guards against trying to access dict fields that don't exist

Reply:
That makes sense and I agree with that suggestion.